lib/semantics/type.cc

```diff
@@ -197,57 +79,43 @@ std::ostream &operator<<(std::ostream &o, const ShapeSpec &x) {
   return o;
 }
 
-std::ostream &operator<<(std::ostream &o, const DataComponentDef &x) {
-  o << x.type_;
-  if (!x.attrs_.empty()) {
-    o << ", " << x.attrs_;
-  }
-  o << " :: " << x.name_.ToString();
-  if (!x.arraySpec_.empty()) {
-    o << '(';
-    int n = 0;
-    for (ShapeSpec shape : x.arraySpec_) {
-      if (n++) {
-        o << ", ";
-      }
-      o << shape;
-    }
-    o << ')';
+IntrinsicTypeSpec::IntrinsicTypeSpec(TypeCategory category, int kind)
+  : category_{category}, kind_{kind ? kind : GetDefaultKind(category)} {
+  CHECK(category != TypeCategory::Derived);
+}
+
+int IntrinsicTypeSpec::GetDefaultKind(TypeCategory category) {
+  switch (category) {
+  case TypeCategory::Character: return evaluate::DefaultCharacter::kind;
+  //case TypeCategory::Complex: return evaluate::DefaultComplex::kind;
+  case TypeCategory::Complex: return 4;  // TEMP to work around bug
```

Comment:
The default kind of COMPLEX will always be the same as the default kind of REAL.